include/zephyr/kernel/thread.h

```diff
@@ -264,6 +264,9 @@ struct k_thread {
 
```

Comment:
Nitpick: can you break the second patch up into one that introduces the new scheduler API and another that uses it in the IPC layer?

Reply:
@andyross addressed this :) Thank you